FirebaseCore/Tests/Unit/Swift/FirebaseAppTests.swift

```diff
@@ -0,0 +1,382 @@
+    app.delete { success in
+      XCTAssertTrue(success)
+      XCTAssertFalse(FirebaseApp.allApps?.keys.contains(self.testApp.testAppName1) ?? false)
+      appDeletedExpectation.fulfill()
+    }
+
+    waitForExpectations()
+  }
+
+  func testGetNameOfDefaultApp() {
+    FirebaseApp.configure()
+
+    // TODO: Use XCTUnwrap after dropping support for Xcode 10
+    guard let defaultApp = FirebaseApp.app() else {
+      return XCTFail("Could not unwrap default app")
+    }
+    XCTAssertEqual(defaultApp.name, testApp.firebaseDefaultAppName)
+  }
+
+  func testGetNameOfApp() throws {
+    XCTAssertNil(FirebaseApp.app(name: testApp.testAppName1))
+
+    let options = FirebaseOptions(googleAppID: testOptions.googleAppID,
+                                  gcmSenderID: testOptions.gcmSenderID)
+    FirebaseApp.configure(name: testApp.testAppName1, options: options)
+
+    // TODO: Use XCTUnwrap after dropping support for Xcode 10
+    guard let app = FirebaseApp.app(name: testApp.testAppName1) else {
+      return XCTFail("Could not unwrap app")
+    }
+    XCTAssertEqual(app.name, testApp.testAppName1)
+  }
+
+  func testOptionsForApp() {
+    FirebaseApp.configure()
+    // TODO: Use XCTUnwrap after dropping support for Xcode 10
+    guard let defaultApp = FirebaseApp.app() else {
+      return XCTFail("Could not unwrap default app")
+    }
+    let defaultOptions = FirebaseOptions.defaultOptions()
+    XCTAssertEqual(defaultApp.options, defaultOptions)
+
+    let options = FirebaseOptions(googleAppID: testOptions.googleAppID,
+                                  gcmSenderID: testOptions.gcmSenderID)
+    let superSecretURLScheme = "com.supersecret.googledeeplinkurl"
+    options.deepLinkURLScheme = superSecretURLScheme
+    FirebaseApp.configure(name: testApp.testAppName1, options: options)
+
+    // TODO: Use XCTUnwrap after dropping support for Xcode 10
+    guard let app = FirebaseApp.app(name: testApp.testAppName1) else {
+      return XCTFail("Could not unwrap app")
+    }
+    XCTAssertEqual(app.name, testApp.testAppName1)
+    XCTAssertEqual(app.options.googleAppID, testOptions.googleAppID)
+    XCTAssertEqual(app.options.gcmSenderID, testOptions.gcmSenderID)
+    XCTAssertEqual(app.options.deepLinkURLScheme, superSecretURLScheme)
+    XCTAssertNil(app.options.androidClientID)
+  }
+
+  func testFirebaseDataCollectionDefaultEnabled() {
+    FirebaseApp.configure()
+
+    // TODO: Use XCTUnwrap after dropping support for Xcode 10
+    guard let app = FirebaseApp.app() else {
+      return XCTFail("Could not unwrap default app")
+    }
+
+    // defaults to true unless otherwise set to no in app's Info.plist
+    XCTAssertTrue(app.isDataCollectionDefaultEnabled)
+
+    app.isDataCollectionDefaultEnabled = false
+    XCTAssertFalse(app.isDataCollectionDefaultEnabled)
+
+    // reset to defautl true since it will persist across runs of the app/tests
+    app.isDataCollectionDefaultEnabled = true
+  }
```

Comment:
Instead of using `configure` for this test - is initializing a custom FirebaseApp sufficient? Or does the implementation depend on `configure` being called?

Reply:
done!